src/www/ui/api/Controllers/SearchController.php

```diff
@@ -88,8 +90,31 @@ public function performSearch($request, $response, $args)
       return $response->withJson($returnVal->getArray(), $returnVal->getCode());
     }
 
+    /*
+     * check if page && pageSize are numeric, if existing
+     */
+    if ((! ($page==='') && (! is_numeric($page) || $page < 1)) ||
```

Comment:
Instead of using `===''`, it is better to use `empty()` as the variable can be `''` or null or 0 or does not exist. The `===` will explicitly check for a string `''` but `empty()` on the other hand will return `true` for all the things mentioned.

Reply:
I have written the code this way since `empty()` cannot differentiate between 0 and does not exist (i.e. parameter not provided by user).  If I use `!empty()`, it will not throw an error for 0. Parameter not existing is handled in a separate `empty()` check where default value is assigned.